astrology.lic

```diff
@@ -23,18 +23,23 @@ class Astrology
       'defensive combat' => 0,
       'future events' => 0
     }
-    Flags.add('rtr-expire', spell['expire']) if Flags['rtr-expire'].nil?
-
+    
     do_buffs(settings)
 
     train_astrology
   end
 
   def do_buffs(settings)
     settings.astrology_buffs['spells'].each do |spell|
-      if spell['abbrev'].casecmp('rtr').zero? 
-        next unless Flags['rtr-expire']
-	Flags.reset['rtr-expire']
+      if spell['abbrev'].casecmp('rtr').zero?
+	case Flags['rtr-expire']
+	when nil
+	  Flags.add('rtr-expire', spell['expire'])
+	when false
+          next
```

Comment:
I just want the flag set on a successful cast. If it fails and still sets the expiration flag, then astrology buffs will never try to cast it again.